nntrainer/src/layer.cpp

```diff
@@ -215,4 +216,20 @@ void Layer::ensureName() {
   }
 }
 
+void Layer::print(std::ostream &out) const {
+  /// @todo change type enum to actual string value
+  out << '<' << typeid(*this).name() << "Object at" << this << '>' << std::endl
+      << "Layer Type: " << type << std::endl
+      << "Activation type: " << activation_type << std::endl
+      << "bias_initiate_to_zero: " << bias_init_zero << std::endl
```

Comment:
not every layer uses `bias` and `weight decay`. Let each layer add this if they use it?

Reply:
I am going to use #270 to filter out valid properties. I guess this should be done in another PR.